tests/firedrake_adjoint/test_reduced_functional.py

```diff
@@ -166,10 +168,11 @@ def test_mixed_boundary():
 def test_assemble_recompute():
     mesh = UnitSquareMesh(10, 10)
     V = FunctionSpace(mesh, "CG", 1)
+    R = FunctionSpace(mesh, "R", 0)
 
     f = Function(V)
     f.vector()[:] = 2
-    expr = Constant(assemble(f**2*dx), domain=mesh)
+    expr = Function(R).assign(assemble(f**2*dx))
```

Comment:
This test fails if I set `Function(R, val=assemble(f**2*dx))`. So, I kept `Function(R).assign(assemble(f**2*dx))`.